packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -363,7 +363,7 @@ class IosProject extends XcodeBasedProject {
     if (globals.cache.isOlderThanToolsStamp(generatedXcodePropertiesFile)) {
       await xcode.updateGeneratedXcodeProperties(
         project: parent,
-        buildInfo: BuildInfo.debug,
+        buildInfo: const BuildInfo(BuildMode.debug, null, trackWidgetCreation: true, treeShakeIcons: false),
```

Comment:
Actually it would be good to do for `BuildInfo.debug` everywhere.  Can you instead update this line?
https://github.com/flutter/flutter/blob/66ed64be4fd8349cf47bf9b86f4d74ae3b4198f8/packages/flutter_tools/lib/src/build_info.dart#L167
See also
https://github.com/flutter/flutter/blob/68471bbb74aa49a2e88606cc5b5f05531ac81e11/packages/flutter_tools/lib/src/runner/flutter_command.dart#L937-L939

@jonahwilliams can you foresee any issues with this?

Reply:
With turning on track-widget-creation? No, I thinks on by default pretty much everywhere, so turn it on here is reasonable too. In general though, we don't have a good way to sync compiler options when using Flutter attach beyond doing a hot restart. This doesn't make the problem any worse, though it does make the default behavior better.

Reply:
Updated.